client/my-sites/domains/domain-management/settings/cards/domain-redirect-card.tsx

```diff
@@ -28,36 +28,40 @@ const noticeOptions = {
 class DomainRedirectCard extends Component {
 	static propTypes = {
 		redirect: PropTypes.object.isRequired,
-		domainName: PropTypes.string.isRequired,
-		targetUrl: PropTypes.string.isRequired,
-		currentRoute: PropTypes.string.isRequired,
+		selectedSite: PropTypes.object.isRequired,
 	};
 
 	state = {
-		redirectUrl: this.props.targetUrl ?? '',
+		targetHost: this.props.redirect.targetHost,
+		secure: this.props.redirect.secure ? 1 : 0,
 	};
 
 	componentDidMount() {
-		this.props.fetchDomainRedirect( this.props.domainName );
+		this.props.fetchDomainRedirect( this.props.selectedSite.domain );
```

Comment:
When a domain is attached to a site it will have the <site>.wordpress.com subdomain in this field.

I could be mistaken but I think we should be avoiding site context for this limiting things to the domain only:

p1691025318606729/1691000360.963039-slack-CKZHG0QCR